### PR TITLE
Update links to the `connectors` repository

### LINKED ISSUE #326: Update links to the archived `connectors` repo
The [connectors](https://github.com/delta-io/connectors) repository has been recently archived (on Jul 13, 2023). Its contents have been moved into the [delta](https://github.com/delta-io/delta) repository, under connectors.

Therefore, any link pointing to the archived repository should be updated to point to the new location instead.

### PR DESCRIPTION
**Pull request by manuteleco**
The [connectors](https://github.com/delta-io/connectors) repository has been recently archived (on Jul 13, 2023). Its contents have been moved into the [delta](https://github.com/delta-io/delta) repository, under `connectors`.

Therefore, any link pointing to the archived repository should be updated to point to the new location instead.

We update links from published blog posts as well, as that's probably the right thing to do.

But we leave untouched any link that refers to an issue, a release, a pull request or a branch, as those should still point to the archived repository. Only links that refer to files or directories within the repository are adjusted.

---

Closes #326

### CODE REVIEW

File: src/blog/2023-07-07-delta-lake-transaction-log-protocol/index.mdx [hunk 1]
```diff
@@ -49,7 +49,7 @@ There are several known open source implementations of Delta Lake:
 - [delta-io/delta](https://github.com/delta-io/delta/)
 - [delta-io/delta-rs](https://github.com/delta-io/delta-rs)
 - [dask-contrib/dask-deltatable](https://github.com/dask-contrib/dask-deltatable)
-- [delta-io/connectors](https://github.com/delta-io/connectors)
+- [delta-io/connectors](https://github.com/delta-io/delta/tree/master/connectors)
```
Comment (manuteleco) [inline src/blog/2023-07-07-delta-lake-transaction-log-protocol/index.mdx:52]: This case is a bit awkward, in that it names the archived repo (delta-io/connectors) but links to the new location. I wonder whether I should simply delete this item instead. What do you think?

Comment (dennyglee) [reply src/blog/2023-07-07-delta-lake-transaction-log-protocol/index.mdx:52]: @manuteleco good point, I think we should just delete it.  Everything looks great.  could you remove it and I'll approve and merge?

**Review by dennyglee — CHANGES_REQUESTED** [review]
Thanks very much for doing this @manuteleco  - everything looks great!  Per your own call out for the delta connectors link, please update and I'll merge this.  Feel free to tag me as well.  Much appreciated!

File: src/blog/2023-07-07-delta-lake-transaction-log-protocol/index.mdx (hunk 1 above)
Comment (manuteleco) [reply src/blog/2023-07-07-delta-lake-transaction-log-protocol/index.mdx]: Removed. Thanks.

**Review by dennyglee — APPROVED** [review]
LGTM!

**Comment by dennyglee** [comment]
Thanks @manuteleco !

**Pull request merged**